plugins/terraform/provider.go

```diff
@@ -1,15 +1,15 @@
 package main
 
 import (
-	"time"
-
 	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/terraform"
 	"github.com/quintilesims/layer0/cli/client"
+	"time"
 )
 
 var defaultTimeout = time.Minute * 15
 
-func Provider() *schema.Provider {
+func Provider() terraform.ResourceProvider {
```

Comment:
It actually fails at that line (19) and at 13 before it when it's `*schema.Provider`:
```
invalid type assertion: Provider().(*schema.Provider) (non-interface type *schema.Provider on left)
```
If I strip out the type assertions, it looks like things work.

I guess it's because `schema.Provider` (_hashicorp/terraform/helper/schema/provider.go_) is type struct, whereas `terraform.ResourceProvider` (_hashicorp/terraform/terraform/resource_provider.go_) is type interface?

Do we have a preference?